controllers/azureidentity_controller.go

```diff
@@ -96,20 +110,36 @@ func (r *AzureIdentityReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		))
 	defer span.End()
 
+	// identityOwner is the resource that created the identity. This could be either an AzureCluster or AzureManagedControlPlane (if AKS is enabled).
+	// check for AzureManagedControlPlane first and if it is not found, check for AzureManagedControlPlane.
+	var identityOwner interface{}
+
 	// Fetch the AzureCluster instance
 	azureCluster := &infrav1.AzureCluster{}
+	identityOwner = azureCluster
 	err := r.Get(ctx, req.NamespacedName, azureCluster)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
+	if err != nil && apierrors.IsNotFound(err) {
+		if feature.Gates.Enabled(feature.AKS) {
+			// Fetch the AzureManagedControlPlane instance
+			azureManagedControlPlane := &infraexpv1.AzureManagedControlPlane{}
+			identityOwner = azureManagedControlPlane
+			err = r.Get(ctx, req.NamespacedName, azureManagedControlPlane)
+			if err != nil && apierrors.IsNotFound(err) {
+				r.Recorder.Eventf(azureCluster, corev1.EventTypeNormal, "AzureClusterObjectNotFound", err.Error())
```

Comment:
(deleted and then decided this was valid, sorry for spam)

err is shadowed, but we know it's not found in both cases. let's either not shadow it and log correct errors, or not log the full error in both cases.